### PR TITLE
Updates CellularNoise to be faster and multi-threaded

### PR DESCRIPTION
**Pull request by EdgeLordExe**
Hello, I guess i'm on a streak. I reimplemented CAnoise to be multi-threaded and much faster as i removed a considerable amount of branches. No change to behaviour or ABI noted, it's just faster now.

obraz

### CODE REVIEW

**Comment by EdgeLordExe** [comment]
Hello could someone review this?

**Comment by AffectedArc07** [comment]
@tgstation/rust-g-dev-team

**Comment by optimumtact** [comment]
there isn't anyone

File: src/cellularnoise.rs [hunk 1]
```diff
@@ -15,76 +16,75 @@ fn noise_gen(
     height_as_str: &str,
 ) -> Result<String> {
     let percentage = percentage_as_str.parse::<i32>()?;
```
Comment (ZeWaka) [inline src/cellularnoise.rs:18]: why not usize this?

File: src/cellularnoise.rs [hunk 2]
```diff
@@ -15,76 +16,75 @@ fn noise_gen(
...
-        let zplane_old = zplane.clone();
-        for i in 0..height {
-            for j in 0..width {
-                let mut sum = 0;
-
-                if i > 0 {
-                    if j > 0 {
-                        sum += if zplane_old[i - 1][j - 1] { 1 } else { 0 };
+    //we populate it
+    let mut filled_vec = (0..height + 3)
```
Comment (ZeWaka) [inline src/cellularnoise.rs:25]: comment explaining the +3s? i get it, but it's not immediately obvious

**Review by ZeWaka — CHANGES_REQUESTED** [review]

File: src/cellularnoise.rs (hunk 2 above)
Comment (EdgeLordExe) [reply src/cellularnoise.rs:25]: good idea

File: src/cellularnoise.rs (hunk 1 above)
Comment (EdgeLordExe) [reply src/cellularnoise.rs:18]: didn't want to touch the API of this, could reasonably change this to usize.

**Comment by optimumtact** [comment]
someone do a release tag

**Comment by ZeWaka** [comment]
> someone do a release tag

go bug moth

**Comment by optimumtact** [comment]
makes you maintainer of rust g

**Pull request merged**